src/protocols/protocol_address_31402.cpp

```diff
@@ -35,19 +35,18 @@ namespace network {
 using namespace bc::message;
 using namespace std::placeholders;
 
-static message::address configured_self(const network::settings& settings)
+static message::address get_addresses(const p2p& network)
 {
-    if (settings.self.port() == 0)
-        return address{};
-
-    return address{ { settings.self.to_network_address() } };
+    bc::message::network_address::list addresses;
+    network.fetch_addresses(addresses);
+    return addresses;
 }
 
 protocol_address_31402::protocol_address_31402(p2p& network,
     channel::ptr channel)
   : protocol_events(network, channel, NAME),
     network_(network),
-    self_(configured_self(network_.network_settings())),
+    self_(get_addresses(network_)),
```

Comment:
This implies we will send the full set of addresses on start() and then the same exact set once if requested, which is of course redundant. It also causes the address set to be stored until the connection drops. The original implementation here is connect (a message of 1 on start). The subsequent send should be from a local/temporary allocation.

Reply:
It's not the full set, it's the randomized subset we want to send.  But agreed that it's redundant, I missed the initial send since originally it just sent our own address.

Reply:
That's what I intended by "full set" in this context. There's no reason at all to store it. The send on start should send just our address (from the configured member), then upon request we respond with a dynamically-generated (and discarded) set. We can do this repeatedly or allow it only once, by simply changing the bool return.